### DISCUSSION TITLE
EPS/Grid CT and Feedin not populating for K series

### DISCUSSION THREAD

**Discussion by williamjeccles** [nathanmarlor/foxess_modbus#533] [Support]
As you can see, since debugging the registers with @canton7 it's mostly working but I've noticed a few values that aren't working.

I'm sure it's just a mapping issue however :)

![Screenshot 2024-02-01 171624](https://github.com/nathanmarlor/foxess_modbus/assets/6324545/35903bcb-65cf-4bf5-86d4-71f35d4888a8)

**Comment by williamjeccles** [comment]
@canton7 let me know if you have a fix you'd like me to test out for this issue. Thanks, Will

**Comment by canton7** [comment]
We're using register 31014, which the KH spec says is "meter1 power":

![image](https://github.com/nathanmarlor/foxess_modbus/assets/568104/57b2e510-79ea-4d80-a8eb-0ad74c00680e)

(That then drives the feed-in and grid consumption sensors).

Interestingly, the other KH spec I have says that register is "TBD":

![image](https://github.com/nathanmarlor/foxess_modbus/assets/568104/ee2bf178-6f57-41a1-9578-811c727b96dd)

 ... and has CT1 up at 31049:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/568104/259d3a5a-d894-49ce-8bf6-ac01521633f2)

Mind reading 31049, see if the value looks sensible?

**Comment by canton7** [comment]
Can you give v1.10.0b6 a try?

**Reply by canton7** [reply]
@williamjeccles would you mind trying the Force Charge / Force Discharge work modes with that version as well please? I've assume srhay the KH supplies some registers, but it's be good to check.

**Reply by williamjeccles** [reply]
@canton7 will do 👍

**Reply by williamjeccles** [reply]
@canton7 looks broken now on that branch 
![image](https://github.com/nathanmarlor/foxess_modbus/assets/6324545/afb5adcf-9eb1-43a6-84d1-7da2d4542780)

**Reply by williamjeccles** [reply]
@canton7 this is the error on the latest build 


```
This error originated from a custom integration.

Logger: custom_components.foxess_modbus.modbus_controller
Source: custom_components/foxess_modbus/modbus_controller.py:258
Integration: FoxESS - Modbus (documentation, issues)
First occurred: 18:55:15 (2 occurrences)
Last logged: 18:55:15

tcp://192.168.1.166:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.HOLDING; start: 44008; count: 5; slave: 247 from tcp://192.168.1.166:502: Exception Response(131, 3, IllegalAddress)
tcp://192.168.1.164:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.HOLDING; start: 44008; count: 5; slave: 247 from tcp://192.168.1.164:502: Exception Response(131, 3, IllegalAddress)
```

**Reply by canton7** [reply]
Aha, that's what I was worried about, thanks -- you don't seem to have access to all of the remote control registers.

Things are a bit slow at the moment as work's hectic, but hopefully that will ease up in the coming couple of weeks.

**Comment by williamjeccles** [comment]
Will continue the discussion in https://github.com/nathanmarlor/foxess_modbus/issues/518
